app/DashBoard/Presenters/HomePagePresenter.php

```diff
@@ -18,20 +20,68 @@ class HomePagePresenter extends BasePresenter
 	private $projectsRepository;
 
 
+	/**
+	 * @var \Pd\Monitoring\UsersFavoriteProjects\UsersFavoriteProjectsRepository
+	 */
+	private $usersFavoriteProjectsRepository;
+
+    /**
+     * @var \Pd\Monitoring\User\User
+     */
+    private $userRepository;
+
+
 	public function __construct(
 		\Pd\Monitoring\DashBoard\Controls\Project\IFactory $projectControlFactory,
-		\Pd\Monitoring\Project\ProjectsRepository $projectsRepository
+		\Pd\Monitoring\Project\ProjectsRepository $projectsRepository,
+        \Pd\Monitoring\UsersFavoriteProjects\UsersFavoriteProjectsRepository $usersFavoriteProjectsRepository,
+        \Pd\Monitoring\User\UsersRepository $userRepository
 	) {
 		$this->projectControlFactory = $projectControlFactory;
 		$this->projectsRepository = $projectsRepository;
+		$this->usersFavoriteProjectsRepository = $usersFavoriteProjectsRepository;
+		$this->userRepository = $userRepository;
 	}
 
-
 	public function renderDefault()
-	{
-		$this->template->projects = $this->projectsRepository->findAll()->orderBy('name');
-	}
+    {
+        $favoriteProjects = $this->usersFavoriteProjectsRepository->findBy(["user" => $this->getUser()->id])->orderBy("this->projects->name");
+
+        $favoriteArray = [];
+        foreach ($favoriteProjects as $favoriteProject) {
+            $favoriteArray[] = $favoriteProject->projects->id;
+        }
+
+        $this->template->projects = $this->projectsRepository->findBy(["id!" => $favoriteArray])->orderBy("name");
+        $this->template->favoriteProjects = $favoriteProjects;
+    }
+
+    public function handleDeleteFavoriteProject(int $id)
+    {
+        $userId = $this->getUser()->id;
+        if($this->usersFavoriteProjectsRepository->checkIfUserHasFavoriteProject($userId, $id)) {
+            $this->usersFavoriteProjectsRepository->deleteFavoriteObject($userId, $id);
+            $this->flashMessage("Oblíbená položka byla odebrána.");
+        } else {
+            $this->flashMessage("Zadaná položka už byla smazána.");
+        }
+        $this->redirect("this");
+    }
 
+    public function handleSetFavoriteProject(int $id)
+    {
+        $userId = $this->getUser()->id;
+        if(!$this->usersFavoriteProjectsRepository->checkIfUserHasFavoriteProject($userId, $id)){
+            $favoriteProject = new UsersFavoriteProjects();
+            $favoriteProject->user = $this->userRepository->findById($userId)->fetch();
+            $favoriteProject->projects = $this->projectsRepository->findById($id)->fetch();
+            $this->usersFavoriteProjectsRepository->persistAndFlush($favoriteProject);
+            $this->flashMessage("Oblíbená položka byla přidána.");
+        } else {
+            $this->flashMessage("Zadaná položka se už v oblíbených nachází.");
+        }
+        $this->redirect("this");
+    }
```

Comment:
Po osobní konzultaci to uděláme bez AJAXU, jen půjde veškerá logika do komponent.